### DISCUSSION TITLE
Sorting default

### DISCUSSION THREAD

**Discussion by ursbraem** [orchidjs/tom-select#200] [Q&A]
We have a  plain select with 2000 options, and a vanilla implementation of tom select:

```
<script>
$(function () {
	var selector = '.js-tom-select select.survey-input',
	  tomSelects = $(selector);

	tomSelects.each(function() {
	  new TomSelect(this, {
	  	maxOptions: 100,
	  });
	});
});
</script>
```

```
<option value="55729273">Bettenhausen</option>
<option value="55729274">Bettens</option>
<option value="55729275">Bettingen</option>
<option value="55729276">Bettlach</option>
```

Now if the user types a single letter, tom select will  mess up the order. 

<img width="294" alt="grafik" src="https://user-images.githubusercontent.com/1446443/132206330-b66cb0a7-2027-4f7f-9e54-39a92bfe95c5.png">

I saw the `sortField` property, also in the basic example at https://tom-select.js.org/examples/, but this produces no effect. Maybe because there are 2k entries to be sorted? Or how do I instruct the plugin to maintain the initial sorting?

**Comment by oyejorge** [comment]
> Or how do I instruct the plugin to maintain the initial sorting?

Don't set sortField https://jsfiddle.net/59r1cnoL/

**Comment by ursbraem** [comment]
Nope, this is my list:

https://jsfiddle.net/ursbraem/0pymckve/3/

To reproduce, type "B"
you will see that the place "Bex" comes before "Baar" which is not in the initial sorting in the html

**Comment by oyejorge** [comment]
Ok, thanks for the clear example. I clearly forgot how the sortField setting works. [Just posted the solution on a similar issue](https://github.com/orchidjs/tom-select/issues/192#issuecomment-914572384)

Essentially, this:

```js
 sortField: [{field:'$order'},{field:'$score'}],
```

**Reply by ursbraem** [reply]
Thanks! Is that correct?

https://jsfiddle.net/ursbraem/rbk9ea0h/4/

As opposed to `[{order:'asc'}]` with `[{field:'$order'},{field:'$score'},{order:'asc'}]` it now filters by any char than just the first. 

(I'm not sure what is best for the use case, just to note the difference. Probably we'd want sth like with 1-2 chars, just filter by word start with more, also filter inside words)

**Comment by oyejorge** [comment]
To specify direction (ascending or descending): ```[{field:'$order',order:'desc'},{field:'$score',...}]```

> Probably we'd want sth like with 1-2 chars, just filter by word start with more, also filter inside words

For that, you'll need a custom sorting function which will be available in the next RC release. [See the sort callback option for sifter](https://github.com/orchidjs/sifter.js#searchquery-options)

**Comment by ursbraem** [comment]
I still get an irregular sorting https://jsfiddle.net/ursbraem/y6fswonp/3/ when typing in one letter (B -> Bex, Baar, Belp), not the wrong direction but mixed up. But I think we can leave it there, tom select is so good overall (we used chosen before which doesn't play along on mobile at all) so that isn't a blocker. 

Thanks Josh!
